### PR TITLE
[23Q1T1] Fade-out Node Glyphs

### PR DESCRIPTION
**Pull request by dnenov**
### Purpose

The first step of implementing the changes to the Color Coding behavior of Dynamo Nodes. This PR adds fade-out behavior to the stack panel containing the visual-cue glyphs (preview/frozen).

Commit 4 - Glyphs Overlay Implementation (2 options for glyph position)

**(preferred)**

glyphs_over

glyphs_under

Commit 3
node_color_overlay
Commit 2
frozen_glyph_implementation
Commit 1
glyph_fade_out

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Updates the visual-cue glyph behavior when zoomed-in/out.

### Reviewers

@sm6srw 

### FYI
@Amoursol

### CODE REVIEW

**Comment by sm6srw** [comment]
Looks good.  Looks like you are close to also enabling the display of the glyphs? Do you want to include that in this PR or in a separate PR. Can you also look into adding a few tests?

**Comment by dnenov** [comment]
Thanks, Jorgen. Only the .xaml part left more or less, let's do it in the same PR. 

I'd love to add tests, but I have two main problems in that regard - a conceptual and a knowledge-based one. I started reading through the existing tests and getting a holistic enough view of the code base to allow me to write a test looks .. a bit intimidating at the moment. On the conceptual level, I would appreciate your support on what should be a well-placed test? I think I only have 1 method so far I'd like tested, but I am not sure if I am right or wrong. 

I will finish up the code and before I get on with the tests, I will maybe chase you up for a quick chat on Slack?

**Comment by sm6srw** [comment]
Great, sounds like a plan. Ping me when you want to chat.

**Review by sm6srw — COMMENTED** [review]
Looks great! Do you feel like we have enough test coverage or do you think we need some tests that checks the properties on the node model?

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml [hunk 1]
```diff
@@ -535,18 +549,90 @@
...
                 Grid.Row="1"
                 Grid.RowSpan="4"
                 Grid.ColumnSpan="3"
                 Margin="-8"
                 Background="{StaticResource NodeFrozenOverlayColor}"
-                Canvas.ZIndex="7"
+                Canvas.ZIndex="6"
+                CornerRadius="8,8,0,0"
+                IsHitTestVisible="False"
+                Opacity="{Binding Path=DataContext.Zoom,
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml:562]: can you verify that in addition to lowering the opacity if the zoom level is high, that you are also making the item collapsed? This should remove the elements from the visual tree, avoiding layout and render costs.

File: test/DynamoCoreWpfTests/NodeViewCustomizationTests.cs [hunk 2]
```diff
@@ -314,9 +314,11 @@ public void WatchImageCoreContainsImage()
 
             // Starting from Dynamo 2.13, node view now comes with 
             // images like node icon, lacing image etc
-            Assert.AreEqual(3, imgs.Count());
+            // As of March 2022, we have 7 images per NodeView
+            // Images are named for ease of use
+            Assert.AreEqual(7, imgs.Count());
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/NodeViewCustomizationTests.cs:319]: any noticeable increase in memory use?

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml (hunk 1 above)
Comment (dnenov) [reply src/DynamoCoreWpf/Views/Core/NodeView.xaml:562]: Sure thing, will do!

Comment (dnenov) [reply src/DynamoCoreWpf/Views/Core/NodeView.xaml:562]: I have introduced the changes, let me know if that hasn't translated into the desired increase in performance.

File: test/DynamoCoreWpfTests/NodeViewCustomizationTests.cs (hunk 2 above)
Comment (sm6srw) [reply test/DynamoCoreWpfTests/NodeViewCustomizationTests.cs:319]: I did a high level comparison watching the process memory in use with and without this change. My use case was to display 792 nodes. Without the change the average was ~1.23 Gb over three runs. ~1.24 Gb over three runs with the change. Looks OK to me.

Comment (dnenov) [reply test/DynamoCoreWpfTests/NodeViewCustomizationTests.cs:319]: Thank you, Jorgen!

**Pull request merged**